src/code/z_fcurve_data.c

```diff
@@ -1,5 +1,84 @@
+/**
+ * File: z_fcurve_data.c
+ * Description: Interpolation functions for use with Curve SkelAnime
+ */
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fcurve_data/func_800F23E0.s")
+#define FCURVE_INTERP_CUBIC 0  // Interpolate using a Hermite cubic spline
+#define FCURVE_INTERP_NONE 1   // Return the value at the left endpoint instead of interpolating
+#define FCURVE_INTERP_LINEAR 2 // Interpolate linearly
```

Comment:
```suggestion
#define FCURVE_INTERP_CUBIC 0  // Interpolate using a Hermite cubic spline
#define FCURVE_INTERP_NONE (1 << 0)   // Return the value at the left endpoint instead of interpolating
#define FCURVE_INTERP_LINEAR (1 << 1) // Interpolate linearly
```
Since these are flags seems appropriate?

Reply:
Perhaps, although the presence of 0, and the impossibility of setting more than one, suggests to me that they aren't "really" flags. It reminds me a lot of the actor allocation stuff in zeldaret/oot#1266 , which is also checked like flags but is really more like an enum. (Of course 0 is not actually checked, but you could do a switch or something if you wanted.)

Reply:
Yeah I'm not sure the best way to handle it myself. They shouldn't be flags, but have flag operations performed on them anyways. How you have them should be fine than.

Reply:
Usage implies enum to me rather than flags.